sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -622,21 +622,43 @@ class SparkSession private(
    *             For example, 1, "Steven", LocalDate.of(2023, 4, 2).
    *             A value can be also a `Column` of literal expression, in that case
    *             it is taken as is.
+   * @param tracker A tracker that can notify when query is ready for execution
    *
    * @since 3.5.0
    */
   @Experimental
-  def sql(sqlText: String, args: Array[_]): DataFrame = withActive {
-    val tracker = new QueryPlanningTracker
-    val plan = tracker.measurePhase(QueryPlanningTracker.PARSING) {
-      val parsedPlan = sessionState.sqlParser.parsePlan(sqlText)
-      if (args.nonEmpty) {
-        PosParameterizedQuery(parsedPlan, args.map(lit(_).expr))
-      } else {
-        parsedPlan
+  def sql(sqlText: String, args: Array[_], tracker: QueryPlanningTracker): DataFrame =
```

Comment:
Looking at these new methods, I don't think that they offer a lot of benefit to outside callers and they are not easy to use. can we make these new methods that have been added `private[sql]`?

@HyukjinKwon @juliuszsompolski what do you think?

Reply:
Yes, I agree they don't need to be exposed as public APIs, but could be internal.

Reply:
Updated to private[sql]